### ISSUE TITLE
Faster BLAS functionality

### ISSUE THREAD

**Issue by mikolalysenko** [scijs/contributing#1]
Currently, we have [ndgemm](https://github.com/scijs/ndgemm) which implements the corresponding BLAS function but it isn't very fast yet.  This code could use some attention/optimizations, or alternatively a rewrite with some better approach.

**Comment by rreusser** [comment]
A benchmark suite, perhaps? In the spirit of modularization, maybe the right answer is to keep it test-driven, modularized, and benchmarked so that the first goal is to get things implemented, after which the components can be swapped with more robust, optimized implementations.

**Comment by mikolalysenko** [comment]
That would be great!

**Comment by rreusser** [comment]
Also, it's not totally clear to me whether the goal should be to move toward porting existing GSL/BLAS-like code via emscripten or whether the goal should be plain vanilla implementations. Oh wait, [this is happening now](https://www.sharelatex.com/github/repos/Planeshifter/GSoC-Application/builds/f7642a61f60249eb0aafad58a1ac05e95ae019aa/raw/output.pdf) :smile:

**Comment by mikolalysenko** [comment]
Yeah, but the proposal ended up getting rejected due to lack of slots.  The jQuery org only got like 3 out of a pool of something like 200 proposals, and so their core projects ended up getting priority over SciJS :(

So probably not gonna happen unfortunately...

**Comment by rreusser** [comment]
Ah, I see. Yeah, wasn't sure if it was accepted but was at least surprised/pleased to see such a recent date. It means there's interest and motivation. I'm more than willing to still try to make things happen one way or another. I promise I'll stop with the talk ASAP and start coding some algorithms…
